File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = curSum = 0
        prefixCount = {0:1}

        for num in nums:
            curSum += num
            j = curSum - k

            res += prefixCount.get(j,0)
            prefixCount[curSum] = 1 + prefixCount.get(curSum,0)
        
        return res

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int length = nums.length, l = 1, r = 1;
        for (r = 1; r<length; r++) {
            if(nums[r] != nums[r-1]) {
                nums[l] = nums[r];
                l++;
            }
        }
        return l;
    }
}